### PR TITLE
fix(playwrighttesting): handled limitation of character in parameters.

### PR DESCRIPTION
**Pull request by kashish2508**
# Contributing to the Azure SDK
 handled limitation of character in parameters.
Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by puagarwa — APPROVED** [review]

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs [hunk 1]
```diff
@@ -53,10 +53,10 @@ public TestRunDto GetTestRun()
                 CloudRunEnabled = false,
                 CiConfig = new CIConfig
                 {
-                    Branch = _cIInfo.Branch,
-                    Author = _cIInfo.Author,
-                    CommitId = _cIInfo.CommitId,
-                    RevisionUrl = _cIInfo.RevisionUrl,
+                    Branch = _cIInfo.Branch?.Length > 500? _cIInfo.Branch?.Substring(0,500): _cIInfo.Branch,
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs:56]: Create an util function to truncate any string as per a given length and use it

**Review by Sid200026 — CHANGES_REQUESTED** [review]
Add tests as well to check truncation is working as expected.

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs [hunk 2]
```diff
@@ -220,6 +220,10 @@ internal class Constants
     internal static readonly string s_playwright_service_reporting_url_environment_variable = "_MPT_REPORTING_URL";
     internal static readonly string s_playwright_service_workspace_id_environment_variable = "_MPT_WORKSPACE_ID";
     internal static readonly string s_playwright_service_auth_type_environment_variable = "_MPT_AUTH_TYPE";
+    internal static readonly string s_playwright_service_runId_Length_exceeded_Error = "The runId cannot be more than 200 characters long";
```
[the remark this answers is not in the document]
Comment (kashish2508) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:223]: using s_playwright_service_runId_length_exceeded_error_message

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs [hunk 3]
```diff
@@ -215,11 +215,15 @@ internal class Constants
...
     internal static readonly string s_invalid_service_endpoint_error_message = "The service endpoint provided is invalid. Please verify the endpoint URL and try again.";
+    internal static readonly string s_playwright_service_runId_length_exceeded_error_message = "The runId cannot be more than 200 characters long";
 
     internal static readonly string s_playwright_service_disable_scalable_execution_environment_variable = "_MPT_DISABLE_SCALABLE_EXECUTION";
     internal static readonly string s_playwright_service_reporting_url_environment_variable = "_MPT_REPORTING_URL";
     internal static readonly string s_playwright_service_workspace_id_environment_variable = "_MPT_WORKSPACE_ID";
     internal static readonly string s_playwright_service_auth_type_environment_variable = "_MPT_AUTH_TYPE";
+
+    // Warning constants
+    internal static readonly string s_playwright_service_runName_truncated_warning = "WARNING: The runName cannot exceed 200 characters. Therefore, it will be truncated to 200 characters.";
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:226]: You can remove the text WARNING

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightReporter.cs [hunk 4]
```diff
@@ -150,16 +150,26 @@ internal void InitializePlaywrightReporter(string xmlSettings)
...
+        }
         var baseUri = new Uri(baseUrl);
         var reporterUtils = new ReporterUtils();
         TokenDetails tokenDetails = reporterUtils.ParseWorkspaceIdFromAccessToken(jsonWebTokenHandler: _jsonWebTokenHandler, accessToken: accessToken);
         var workspaceId = tokenDetails.aid;
-
+        var runNameString = runName?.ToString();
+        if (runNameString?.Length > 200)
+        {
+            runNameString = runNameString.Substring(0, 200);
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightReporter.cs:166]: Use truncate util function

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs [hunk 5]
```diff
@@ -104,16 +104,16 @@ public TestResults GetTestCaseResultData(TestResult? testResultSource)
             };
             testCaseResultData.TestCombinationId = testCaseResultData.TestExecutionId; // TODO check
             testCaseResultData.TestId = testResultSource.TestCase.Id.ToString();
-            testCaseResultData.TestTitle = testResultSource.TestCase.DisplayName;
+            testCaseResultData.TestTitle = testResultSource.TestCase.DisplayName.Length > 500? testResultSource.TestCase.DisplayName.Substring(0, 500): testResultSource.TestCase.DisplayName;
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs:107]: why not use the truncate function directly?

**Review by Sid200026 — APPROVED** [review]

**Pull request merged**